addons/zylann.editor_debugger/plugin.gd

```diff
@@ -1,25 +1,44 @@
 @tool
 extends EditorPlugin
 
-var _dock = null
+const Dock := preload("res://addons/zylann.editor_debugger/dock.gd")
+const DockScene: PackedScene = preload("res://addons/zylann.editor_debugger/dock.tscn")
+var _dock: Dock = null
 
 
-func _enter_tree():
-	_dock = load("res://addons/zylann.editor_debugger/dock.tscn")
-	_dock = _dock.instantiate()
+func _enter_tree() -> void:
+	_dock = DockScene.instantiate()
 	_dock.node_selected.connect(_on_EditorDebugger_node_selected)
+	_dock.get_theme_icon = Callable(self, "get_theme_icon")
 	add_control_to_dock(DOCK_SLOT_RIGHT_UL, _dock)
 	
-	#var editor_settings = get_editor_interface().get_editor_settings()
+	#var editor_settings := get_editor_interface().get_editor_settings()
 
 
-func _exit_tree():
+func _exit_tree() -> void:
 	remove_control_from_docks(_dock)
 	_dock.free()
 	_dock = null
 
 
-func _on_EditorDebugger_node_selected(node):
+func _on_EditorDebugger_node_selected(node: Node) -> void:
 	if _dock.is_inspection_enabled():
 		# Oops.
 		get_editor_interface().inspect_object(node)
+
+
+# The default "icon not found" texture. Captured so it can be compared against when trying to
+# find a specific icon
+# @see get_theme_icon
+var _no_texture := get_editor_interface().get_base_control().get_theme_icon("", "EditorIcons")
+
+
+# Returns a theme icon for the provided class name. You may provide an 
+# alternative class name as the second argument to avoid getting a "not found" icon
+# @param icon_name              For example, "Button", or "Node2D"
+# @param if_not_found_pick_this Generally, "Node". Leave blank to get a broken "not found" icon
+func get_theme_icon(icon_name: String, if_not_found_pick_this:="") -> Texture2D:
```

Comment:
It's required to use `get_editor_interface`,  which is the only reliable way I know to obtain editor icons for the nodes and such.

Reply:
Can't controls present in the editor tree access icons? I'd rather just having the thing there rather than being injected. The dock is a child of the editor interface so it should be able to access icons.

Reply:
It should be possible yes. I'll test it.

Reply:
Turns out, `get_theme_icon` is a method of `Control`. I'm unsure if it wasn't in 3.x and it changed, or if I've simply been doing this delegation dance for nothing all along.